crates/subspace-farmer/src/bin/subspace-farmer/bench_rpc_client.rs

```diff
@@ -32,114 +26,53 @@ pub struct Inner {
             async move {
-                let mut segment_index = 0;
-                let mut last_archived_block = LastArchivedBlock {
-                    number: 0,
-                    archived_progress: ArchivedBlockProgress::Partial(0),
-                };
-                loop {
-                    last_archived_block
-                        .archived_progress
-                        .set_partial(segment_index as u32);
-
-                    let archived_segment = {
-                        let root_block = RootBlock::V0 {
-                            segment_index,
-                            records_root: Sha256Hash::default(),
-                            prev_root_block_hash: Sha256Hash::default(),
-                            last_archived_block,
-                        };
-
-                        let mut pieces = FlatPieces::new(100);
-                        rand::thread_rng().fill(pieces.as_mut());
-
-                        let objects = std::iter::repeat_with(|| PieceObject::V0 {
-                            hash: rand::random(),
-                            offset: rand::random(),
-                        })
-                        .take(100)
-                        .collect();
-
-                        ArchivedSegment {
-                            root_block,
-                            pieces,
-                            object_mapping: vec![PieceObjectMapping { objects }],
-                        }
-                    };
-
-                    if archived_segments_sender
-                        .send(archived_segment)
-                        .await
-                        .is_err()
-                    {
+                while let Some(segment) = outer_archived_segments_receiver.recv().await {
+                    if archived_segments_sender.send(segment).await.is_err() {
                         break;
                     }
                     if acknowledge_archived_segment_receiver.recv().await.is_none() {
                         break;
                     }
-
-                    segment_index += 1;
-                    writen_piece_count.fetch_add(100, Ordering::AcqRel);
                 }
             }
         });
 
-        Self {
+        let me = Self {
             inner: Arc::new(Inner {
                 metadata,
                 slot_info_receiver: Arc::new(Mutex::new(slot_info_receiver)),
                 archived_segments_receiver: Arc::new(Mutex::new(archived_segments_receiver)),
                 acknowledge_archived_segment_sender,
                 slot_info_handler: Mutex::new(slot_info_handler),
                 segment_producer_handle: Mutex::new(segment_producer_handle),
-                writen_piece_count,
             }),
-        }
-    }
-
-    pub fn writen_piece_count(&self) -> u64 {
-        self.inner.writen_piece_count.load(Ordering::Relaxed)
+        };
+        (me, outer_archived_segments_sender, outer_slot_info_sender)
```

Comment:
It could have received receiver side of the channel instead, then there would be no need to return multiple things here.